### PR TITLE
EDGAR: Issue 132 use client secret and client id from setup string

### PR DESCRIPTION
**Pull request by mtwardawski**

### CODE REVIEW

File: opendut-edgar/src/setup/tasks/write_configuration.rs [hunk 1]
```diff
@@ -158,6 +184,12 @@ mod tests {
             config_override: ConfigOverride {
                 peer_id: new_peer_id,
                 carl_url: Url::parse("https://example.com:1234").unwrap(),
+                auth_config: AuthConfig {
+                    issuer_url: Url::parse("https://test.com:1234").unwrap(),
+                    client_secret: ClientSecret(CLIENT_SECRET.to_string()),
+                    client_id: ClientId(CLIENT_ID.to_string()),
```
Comment (reimarstier) [inline opendut-edgar/src/setup/tasks/write_configuration.rs:190]: Add serialization for auth config when OIDC enabled/disabled.

**Review by reimarstier — CHANGES_REQUESTED** [review]
As discussed earlier I'd like to change the authentication config to indicate when the authentication is disabled via a rust enum.

File: opendut-types/src/util/net.rs [hunk 2]
```diff
@@ -409,4 +409,9 @@ impl AuthConfig {
             scopes: vec![],
         }
     }
+
+    // TODO: This should be enhanced or refactored, once OIDC and Authentication is fully implemented. Also this may not be the right place for these functions at all.
+    pub fn is_oidc_enabled(&self) -> bool {
```
[the remark this answers is not in the document]
Comment (mtwardawski) [reply opendut-types/src/util/net.rs]: As we discussed, we switched to a enum.

File: opendut-edgar/edgar.toml [hunk 3]
```diff
@@ -15,11 +15,11 @@ ca = "/etc/opendut/tls/ca.pem"
 domain.name.override = ""
 
 [network.oidc]
-enabled = false
+enabled = true
```
[the remark this answers is not in the document]
Comment (mtwardawski) [reply opendut-edgar/edgar.toml]: Changed back

File: opendut-edgar/src/setup/tasks/write_configuration.rs (hunk 1 above)
Comment (mtwardawski) [reply opendut-edgar/src/setup/tasks/write_configuration.rs]: Also changed the implementation to what you suggestion.

File: opendut-edgar/src/setup/tasks/write_configuration.rs [hunk 4]
```diff
@@ -136,16 +168,69 @@ mod tests {
...
+
+    #[test]
+    fn should_write_a_fresh_configuration_with_auth_config_enabled() -> anyhow::Result<()> {
+        let temp = TempDir::new()?;
+        let config_file = temp.child("edgar.toml");
+        let config_merge_suggestion_file = temp.child("edgar-merge-suggestion.toml");
+
+        let new_peer_id = PeerId::from(uuid!("dc72f6d9-d700-455f-8c31-9f15438e7503"));
+
+        let task = WriteConfiguration {
```
Comment (reimarstier) [inline opendut-edgar/src/setup/tasks/write_configuration.rs:191]: There are multiple duplicate code warnings for this file. What about creating a test fixture for those shared parts?

Comment (reimarstier) [reply opendut-edgar/src/setup/tasks/write_configuration.rs:191]: Use the `#[fixture]` macro from rstest_macros to create a struct for structs that are used multiple times.

**Review by reimarstier — CHANGES_REQUESTED** [review]

File: opendut-edgar/src/setup/tasks/write_configuration.rs (hunk 4 above)
Comment (reimarstier) [reply opendut-edgar/src/setup/tasks/write_configuration.rs]: Great. This reduces repetition and now you have used fixtures in rust as well! :-)

**Pull request merged**